AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m

```diff
@@ -87,13 +87,8 @@ - (void)removeDelegate:(id<MSACChannelDelegate>)delegate {
 }
 
 - (void)enumerateDelegatesForSelector:(SEL)selector withBlock:(void (^)(id<MSACChannelDelegate> delegate))block {
-  NSArray *synchronizedDelegates;
-  @synchronized(self) {
-
-    // Don't execute the block while locking; it might be locking too and deadlock ourselves.
-    synchronizedDelegates = [self.delegates allObjects];
-  }
-  for (id<MSACChannelDelegate> delegate in synchronizedDelegates) {
+  NSArray *delegatesCopy = [self.delegates.allObjects copy];
```

Comment:
Is it safe to remove `@synchronized(self)`? What would happen if we call `enumerateDelegatesForSelector` at the same time from multiple threads?

Reply:
I don't think we need it here. I simple don't understand why should we assign it under synchronized. If we call enumerateDelegatesForSelector at the same time from multiple threads nothing bad should happen i think, maybe im wrong